### PR TITLE
Issue #2990876: Refactor the field plugin class and configuration.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: modules/geocoder_field/geocoder_field.module [hunk 1]
```diff
@@ -82,10 +82,9 @@ function geocoder_field_form_field_config_edit_form_alter(array &$form, FormStat
  *   The form state object.
  */
 function geocoder_field_field_config_edit_form_validate(array $form, FormStateInterface $form_state) {
-  // Don't store any settings for this field if it's not configured to use
-  // geocoding.
+  // Don't store any settings for this field if it's not configured.
   $third_party_settings = $form_state->getValue('third_party_settings');
-  if ($third_party_settings['geocoder_field']['method'] == 'none') {
+  if ('none' === $third_party_settings['geocoder_field']['method']) {
```
Comment (pfrenssen) [inline modules/geocoder_field/geocoder_field.module]: The de facto standard in Drupal is to not use Yoda conditions.

File: modules/geocoder_field/geocoder_field.module [hunk 2]
```diff
@@ -151,126 +151,120 @@ function geocoder_field_entity_presave(EntityInterface $entity) {
     }
 
     $geocoder = $field_config->getThirdPartySettings('geocoder_field');
-    if (empty($geocoder['method']) || $geocoder['method'] === 'none') {
+    if (empty($geocoder['method']) || 'none' === $geocoder['method']) {
```
Comment (pfrenssen) [inline modules/geocoder_field/geocoder_field.module]: Please no Yoda conditions. This is not Symfony code.

File: modules/geocoder_field/geocoder_field.module (hunk 1 above)
Comment (drupol) [reply modules/geocoder_field/geocoder_field.module]: Fixed.

File: modules/geocoder_field/geocoder_field.module (hunk 2 above)
Comment (drupol) [reply modules/geocoder_field/geocoder_field.module]: Fixed.

**Comment by itamair** [comment]
While testing I found this matter: 
https://www.drupal.org/project/geocoder/issues/2990876#comment-12721092

**Pull request merged**